File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(number)
    responses = {"is_factor_of_three": "Pling", "is_factor_of_five": "Plang", "is_factor_of_seven": "Plong", "not_a_factor": "#{number}"}
    result = ""

    result << responses[:is_factor_of_three] if number % 3 == 0
    result << responses[:is_factor_of_five] if number % 5 == 0
    result << responses[:is_factor_of_seven] if number % 7 == 0
    return responses[:not_a_factor] if result.empty?
    result
  end
end

File: python/raindrops/raindrops.py
def raindrops(number):
    if number % 3 == 0:
        result = 'Pling'
    elif number % 5 == 0:
        result = 'Plang'
    elif number % 7 == 0:
        result = 'Plong'
    else:
        result = str(number)
    return result
